packages/core/src/utils.js

```diff
@@ -665,12 +665,40 @@ export function stubExistingAdditionalProperties(
   return schema;
 }
 
+const resolveCondition = (schema, rootSchema, formdata) => {
+  let {
+    if: expression,
+    then,
+    else: otherwise,
+    ...resolvedSchemaLessConditional
+  } = schema;
+
+  const conditionalSchema = isValid(expression, formdata, rootSchema)
+    ? then
+    : otherwise;
+
+  if (conditionalSchema) {
+    return retrieveSchema(
+      mergeSchemas(
+        resolvedSchemaLessConditional,
+        retrieveSchema(conditionalSchema, rootSchema, formdata)
```

Comment:
Noticed that this was swapped (presumably to fix a bug?). Is there a test case we could add that would catch a regression?

Reply:
```
{
if: ...,
then: {minLength: 5},
minLength: 10
}
```

Expected behavior: inner property (`minLength: 5`) should override outer property (`minLength: 10`). We should add a test case for this